akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -941,35 +940,69 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
   /**
    * Initiates a new round of gossip.
    */
-  def gossip(): Unit = {
-
+  def gossip(): Unit =
     if (!isSingletonCluster) {
-      val localGossip = latestGossip
+      val gossipTargets =
+        // TODO break this two methods out and unit test?
+        if (latestGossip.isMultiDc) multiDcGossipTargets()
+        else localDcGossipTargets()
+
+      selectRandomNode(gossipTargets) match {
```

Comment:
I was thinking about something like that but then I thought about the case where all those nodes are unavailable and having to "try again" with another DC which the current scheme avoids.  With the "many more nodes" it is still limited by the max-connection setting so as long as we choose a fairly low number for that it wouldn't make a big difference. 

I agree on the 80/20 though. Will add that.